sympy/functions/elementary/tests/test_piecewise.py

```diff
@@ -52,6 +52,13 @@ def test_piecewise1():
     assert Piecewise((1, x > 0), (2, And(x <= 0, x > -1))
         ) == Piecewise((1, x > 0), (2, x > -1))
 
+    # test for supporting Contains in Piecewise
+    assert Piecewise(
+        (S(1), And(Le(x, 6), Ge(x, 1), S.Integers.contains(x))),
+        (S(0), True)) == Piecewise(
+                        (1, Contains(x, S.Integers) & (x >= 1) & (x <= 6)),
+                        (0, True))
```

Comment:
This is a tautological assertion. You should use unchanged for this test. Also we want to test that you can substitute into the Piecewise.

Reply:
The logic used by`unchanged` is incorrect, I am fixing it. 
Take the following for example,

**Code of `unchanged`**
```python
f = func(*args)
return f.func == func and f.args == tuple([sympify(a) for a in args])
```

The above will not work for `Piecewise` as,
```python
cond = And(Le(x, 6), Ge(x, 1), S.Integers.contains(x))
pargs = ((S(1), cond), (S(0), True))
p2 = Piecewise(*pargs)
# import inspect
# print(inspect.getmodule(Contains))
print("type of args in Piecewise -> ", type(p2.args[0]))
print("Actual type of args -> ", type(sympify(pargs[0])))
```
gives,
```python
type of args in Piecewise ->  <class 'sympy.functions.elementary.piecewise.ExprCondPair'>
Actual type of args ->  <class 'sympy.core.containers.Tuple'>
```